Lecture slides at https://www.uio.no/studier/emner/matnat/math/STK4900/v21/lectures/lecture1.pdf

---------------------------

# Exercise 2:  two-sample t-test and confidence interval

In [1]:
#At the lectures we looked an example on bone mineral density (cf. slide 25 from the lectures)

#We will in this exercise see how the computations for this example may be done in R.

 

 

#Start by reading the data into R. This may be done by the command:

cont=c(0.228, 0.207, 0.234, 0.220, 0.217, 0.228, 0.209, 0.221, 0.204, 0.220, 0.203, 0.219, 0.218, 0.245, 0.210)

treat=c(0.250, 0.237, 0.217, 0.206, 0.247, 0.228, 0.245, 0.232, 0.267, 0.261, 0.221, 0.219, 0.232, 0.209, 0.255)

In [23]:
# Find the means and standard deviations, and check that you get the same results as in the lectures (slide 25)

mean_c = mean(cont)
#std_c = (sum((cont-mean_c)**2) / (length(cont)-1))**(1/2)      # true definition
std_c = sd(cont)


mean_t = mean(treat)
std_t = sd(treat)


In [27]:
round(mean_c, 4)
round(std_c, 4)

[1] 0.2189

[1] 0.0116

In [28]:
round(mean_t, 4)
round(std_t, 4)

[1] 0.2351

[1] 0.0188

We see that the outputs coincide with the lecture notes.

In [30]:
# Use the command "t.test" to compute the confidence interval, t-statistic and P-value:

t.test(treat, cont, var.equal=T)


	Two Sample t-test

data:  treat and cont
t = 2.8442, df = 28, p-value = 0.008227
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.004532839 0.027867161
sample estimates:
mean of x mean of y 
0.2350667 0.2188667 


Confidence interval: [0.0045, 0.0279] =  0.0162 $\pm$ 0.0117\
t-statistic: 2.8442 \
p-value: 0.0082

Make sure that you understand the output from the command and check that you get the same results as in the lectures (slides 27 and 28) \
We do get the same values as in the lecture notes.